data_resource_api/app/exception_handler.py

```diff
@@ -0,0 +1,92 @@
+import json
+from brighthive_authlib import OAuth2ProviderError
+from data_resource_api.logging import LogFactory
+from flask import jsonify
+from werkzeug.exceptions import NotFound
+
+
+class DRApiError(Exception):
+    def __init__(self, message, status_code=400, errors=[]):
+        Exception.__init__(self)
+        self.message = message
+        self.errors = errors
+        self.status_code = status_code
+
+    def get_message(self):
+        resp = {}
+        resp['error'] = self.message
+
+        if len(self.errors) != 0:
+            resp['errors'] = self.errors
+
+        return jsonify(resp)
+
+    def get_status_code(self):
+        return self.status_code
+
+
+class ApiError(DRApiError):
+    """Returns an error to the client and does not log the exception.
+    """
+    pass
+
+
+class ApiErrorLog(DRApiError):
```

Comment:
Okaaaay. I do not like name of this class, as it stands, and I see two options:

1. EASY – rename it to something that denotes the error handles "bare" exceptions and things-that-BrightHive-does-not-have-a-custom-error-for, e.g., `ApiUnhandledError` or `ApiUnknownError` or `ApiNewFangledError`.
2. NOT AS EASY – integrate the logging facility into the class. I am not sure how to do this....maybe something like:

```python
class ApiErrorLog(DRApiError):
    def log_error:
        # Or instantiate the logger outside of the class, and pass it in an `init` function?
        logger = LogFactory.get_console_logger('data-model-manager')
        logger.exception("Encountered an error while processing a request.")

....
# below
if isinstance(e, ApiError):
    ApiError.log_error()
    return e.get_message(), e.get_status_code()
```

^^^ entirely untested! So, it might be a fruitless suggestion.

Reply:
Renamed to ApiUnhandledError